File: CSC110-Cpp/Module07-Writing-Classes/debug01/rectangle.cpp
// rectangle.cpp
// A Rectangle class with a constructor, getters, and area/perimeter methods,
// plus a short test program that reads the dimensions from the user.
#include <iostream>
using namespace std;

class Rectangle {
private:
    double width = 0;
    double height = 0;

public:
    Rectangle(double width, double height) {
        width = width;
        height = height;
    }

    double getWidth() { return width; }
    double getHeight() { return height; }

    double area() { return width * height; }
    double perimeter() { return 2 * (width + height); }
};

int main() {
    double w, h;

    cout << "Enter width and height: ";
    cin >> w >> h;

    Rectangle room(w, h);

    cout << "Width:     " << room.getWidth() << endl;
    cout << "Height:    " << room.getHeight() << endl;
    cout << "Area:      " << room.area() << endl;
    cout << "Perimeter: " << room.perimeter() << endl;
    return 0;
}

File: CSC110-Java/_solutions/Module07-Writing-Classes/debug01/Rectangle.java
// Rectangle.java
// Represents a rectangle with a width and a height, both in inches.

public class Rectangle {
    private double width;
    private double height;

    public Rectangle(double width, double height) {
        this.width = width;
        this.height = height;
    }

    public double getWidth() {
        return width;
    }

    public double getHeight() {
        return height;
    }

    public void setWidth(double newWidth) {
        width = newWidth;
    }

    public void setHeight(double newHeight) {
        height = newHeight;
    }

    public double area() {
        return width * height;
    }

    public double perimeter() {
        return 2 * (width + height);
    }
}
